src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,183 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    public static final String FILEPATH = "./shimaStorage.txt";
+    public static final String DELIMITER = "Ø";
+    public static File file = new File(FILEPATH);
+    public static String STORAGE_MESSAGE = "Welcome to my storage :P, this is how I memorize all your tasks!\n" +
+            "Alert! Please do not delete anything inside this file, else I will get memory loss :(\n";
+
+    /**
+     * Creates the storage file
+     *
+     * @throws IOException Throws this exception when error occurs during the write file process
+     */
+    public static void createFile() throws IOException {
+        FileWriter createFile = new FileWriter(FILEPATH);
+        createFile.write(STORAGE_MESSAGE);
+        createFile.close();
+    }
+
+    /**
+     * Reads the data from the storage file
+     *
+     * @param tasks The array list that stores all the tasks
+     * @throws DukeException.StorageException Throws this exception when the data stored in the storage file is invalid
+     * @throws IOException                    Throws this exception when error occurs during the write file process
+     */
+    public static void readFromStorage(ArrayList<Task> tasks) throws DukeException.StorageException, IOException {
+        try {
+            Scanner sc = new Scanner(Storage.file);
+            //Skips the first two lines of storage descriptions in the file
+            int skipTwoLines = 0;
+            while (sc.hasNext()) {
+                if (skipTwoLines < 2) {
+                    sc.nextLine();
+                    skipTwoLines++;
+                } else {
+                    //Starts reading data
+                    String[] tasksData = sc.nextLine().split(DELIMITER);
+                    Task currentTask;
+                    switch (tasksData[0]) {
+                    case "T":
+                        tasks.add(new ToDo(tasksData[2]));
+                        currentTask = tasks.get(tasks.size() - 1);
+                        //Updates the longestTaskDescription to ensure that the frames for to-do list can be printed correctly
+                        Duke.longestTaskDescription = Math.max(currentTask.getTask().length(), Duke.longestTaskDescription);
+                        break;
+                    case "D":
+                        tasks.add(new Deadline(tasksData[2], tasksData[3]));
+                        currentTask = tasks.get(tasks.size() - 1);
+                        //Updates the Duke.longestTaskDescription to ensure that the frames for to-do list can be printed correctly
+                        Duke.longestTaskDescription = Math.max(currentTask.getTask().length() + "(by: )".length() + currentTask.getTime().length(), Duke.longestTaskDescription);
```

Comment:
This line may be too long. Note that it is also affected by the deep indentation. Similarly for all the lines >120 characters, including indentations.